src/env/node/git/localGitProvider.ts

```diff
@@ -1240,6 +1250,57 @@ export class LocalGitProvider implements GitProvider, Disposable {
 		await this.git.branch(repoPath, '-m', oldName, newName);
 	}
 
+	@log()
+	async deleteBranch(
```

Comment:
Nitpick: since this takes an array of branches as a param, shouldn't we call this function `deleteBranches`? Same with the provider service function for consistency, though that one can take either a single or multiple branches.

Reply:
Hey @axosoft-ramint ! Sure, let me go over your questions:
> 1. Describing how to initiate the commands being changed (all the different ways we can get into the delete branch flow that you are aware of)
This behavior hasn't changed. I usually trigger the delete branch action from the context menu in the graph. But since the context menu opens the command palette, you can directly execute the `GitLens: Git Delete Branch` action from the command palette as well.
Note that if you do the latter you will be asked to select the branch since it has no context.

> 2. Describing what happens before this PR and what happens after the PR changes (essentially, what behavior should I expect to be different when using this PR)
> 3. Describing what areas may have regressions as a result of the PR changes that I should spot-check, if any

Before this PR, executing the git action would run a terminal with the commands needed to delete the branch. 
See the screenshot, where I ran a create branch action to create a branch called `test_branch` and then I ran a delete branch action to delete the same branch:
![Captura de pantalla 2024-10-01 a las 10 45 22](https://github.com/user-attachments/assets/2fa0cda0-3e0e-4018-b842-95692fb0e46f)

However, now we run the commands through our git binary and wrap any possible error. See the following example (it was forced to have a bad name, this is not possible because we filter invalid characters out):
![Captura de pantalla 2024-09-30 a las 16 10 45](https://github.com/user-attachments/assets/9c235ca6-ea8f-40af-adb3-ea1be6454e40)

So the expected behavior is that when a `git branch` command is run (create, rename or delete), it doesn't open a terminal anymore and performs the action. If anything happens, it should display a toast with the error information.


> Nitpick: since this takes an array of branches as a param, shouldn't we call this function `deleteBranches`? Same with the provider service function for consistency, though that one can take either a single or multiple branches.

Yeah makes sense. I followed the naming convention of `createBranch` and `renameBranch` thinking of the `git branch` command, so I would group them all under the `branch` name if that makes sense. But it makes more sense for a method that may take multiple branches to be called `deleteBranches` instead. Good catch!

Reply:
The naming is one issue here, and maybe it should be broken into 2 methods, a `deleteBranch` and `deleteBranches`, but imo the bigger issue is the error handling. The first error breaks the whole operation. Unless we want to create a good AggregateError model so that a caller can handle all the issue, I would suggest that we leave the deleting multiple branches up to the calling code (we can wrap it in its own function to be shared -- in `git/actions/branch` probably), and just have the method here deal with a single branch (and optionally its remote) so that errors can be handled, prompted and maybe resumes independently.